docs/serialisation/json.md

```diff
@@ -110,3 +110,45 @@ are encoded as such
 { "tpe": "first", "myString": "alloy" }
 { "tpe": "second", "myInt": 42 }
 ```
+
+### Null values
+
+By default, in Smithy a field having a value of `null` and the field being absent are not distinguished. However, this is not required by JSON semantics and APIs can treat the two differently. In order to support this, the additional trait `alloy.nullable` is provided. Annotating the member of a structure field with this indicates that explicit nulls should be preserved on both serialization and deserialization.
+
+For example, assuming the following smithy structure
+
+```smithy
+use alloy#nullable
+
+structure Foo {
+  @nullable
+  nullable: Integer
+  regular: Integer
+}
+```
+
+The JSON objects
+
+```json
+{ "nullable": null, "regular": null }
+{ "nullable": 4, "regular": 4 }
+{}
+```
+
+are respectively decoded as
+
+```scala
+Foo(Some(Nullable.Null), None)
+Foo(Some(Nullable.Value(4)), Some(4))
+Foo(None, None)
+```
+
+or some similar type which preserves the information that an explicit `null` was passed. These objects are in turn encoded as
```

Comment:
I struggled a little with this example, since my understanding is that the docs are also for protocol implementers and shouldn't assume a `smithy4s`-based code generation, but on the other hand the description of `nullable` makes much more sense if you see the corresponding Scala code as well. Let me know if you think this should be worded differently.

Reply:
I'm not completely opposed to it but the tooling involved in such examples must be clarified